Network/DNS/Resolver.hs

```diff
@@ -450,14 +443,14 @@ lookupRawInternal rcv ad rlv dom typ = loop (NE.uncons (dnsSocks rlv))
 -- quickly on the first try.  There will be no further retries.
 
 tcpRetry :: ByteString
-         -> Socket
+         -> AddrInfo
          -> Int
          -> IO (Either DNSError DNSMessage)
-tcpRetry query sock tm = do
-    peer <- getPeerName sock
-    bracket (tcpOpen peer)
+tcpRetry query ai tm = do
+    let addr = addrAddress ai
+    bracket (tcpOpen $ addr)
```

Comment:
I should probably drop the extraneous `$` before this is merged...

Reply:
Yeah.

Reply:
Fixed in the next commit.